main.cpp

```diff
@@ -163,98 +171,414 @@ good to go!
+    FloatType& multiply( const FloatType& ft );
+    FloatType& divide( const FloatType& ft );
+    
+    FloatType& add( const DoubleType& dt );
+    FloatType& subtract( const DoubleType& dt );
+    FloatType& multiply( const DoubleType& dt );
+    FloatType& divide( const DoubleType& dt );
+    
+    FloatType& add( const IntType& it );
+    FloatType& subtract( const IntType& it );
+    FloatType& multiply( const IntType& it );
+    FloatType& divide( const IntType& it );
 };
+
 struct DoubleType
 {
-    double add(double lhs, double rhs);
-    double subtract(double lhs, double rhs);
-    double multiply(double lhs, double rhs);
-    double divide(double lhs, double rhs);
+    
+    double* valuePtr;
+    DoubleType(float dvalue) : valuePtr(new double(dvalue)) {}
+    ~DoubleType(){delete valuePtr; valuePtr = nullptr;}
+
+    DoubleType& add(double rhs );
+    DoubleType& subtract(double rhs );
+    DoubleType& multiply(double rhs );
+    DoubleType& divide(double rhs );
+    
+    DoubleType& add( const FloatType& ft );
+    DoubleType& subtract( const FloatType& ft );
+    DoubleType& multiply( const FloatType& ft );
+    DoubleType& divide( const FloatType& ft );
+    
+    DoubleType& add( const DoubleType& dt );
+    DoubleType& subtract( const DoubleType& dt );
+    DoubleType& multiply( const DoubleType& dt );
+    DoubleType& divide( const DoubleType& dt );
+    
+    DoubleType& add( const IntType& it );
+    DoubleType& subtract( const IntType& it );
+    DoubleType& multiply( const IntType& it );
+    DoubleType& divide( const IntType& it );
 };
+
 struct IntType
 {
-    int add(int lhs, int rhs);
-    int subtract(int lhs, int rhs);
-    int multiply(int lhs, int rhs);
-    int divide(int lhs, int rhs);
+    int* valuePtr;
+    IntType(int ivalue) : valuePtr(new int(ivalue)) {}
+    ~IntType(){delete valuePtr; valuePtr = nullptr;}
+
+    IntType& add(int rhs );
+    IntType& subtract(int rhs );
+    IntType& multiply(int rhs );
+    IntType& divide(int rhs );
+    
+    IntType& add( const FloatType& ft );
+    IntType& subtract( const FloatType& ft );
+    IntType& multiply( const FloatType& ft );
+    IntType& divide( const FloatType& ft );
+    
+    IntType& add( const DoubleType& dt );
+    IntType& subtract( const DoubleType& dt );
+    IntType& multiply( const DoubleType& dt );
+    IntType& divide( const DoubleType& dt );
+    
+    IntType& add( const IntType& it );
+    IntType& subtract( const IntType& it );
+    IntType& multiply( const IntType& it );
+    IntType& divide( const IntType& it );
 };
 
-float FloatType::add(float lhs, float rhs)
+FloatType& FloatType::add(float rhs)
+{
+    *valuePtr += rhs;
+    return *this;
+}
+
+FloatType& FloatType::subtract(float rhs)
+{
+    *valuePtr -= rhs;
+    return *this;
+}
+
+ FloatType& FloatType::multiply(float rhs)
+{
+    *valuePtr *= rhs;
+    return *this;
+}
+
+FloatType& FloatType::divide(float rhs)
+{
+    *valuePtr /= rhs;
+    return *this;
+}
```

Comment:
The previous project version issued a warning here.  Where is your warning?
- [ ] revise